packages/piro/src/Piro_TempusIntegrator_Def.hpp

```diff
@@ -107,192 +107,188 @@ Piro::TempusIntegrator<Scalar>::TempusIntegrator(Teuchos::RCP< Teuchos::Paramete
   if (adjSensIntegrator_ != Teuchos::null) {
-    xdot = adjSensIntegrator_->getXDot(); 
+    xdot = adjSensIntegrator_->getXDot();
   }
-  return xdot; 
+  return xdot;
 }
 
 template <typename Scalar>
-Teuchos::RCP<const Thyra::VectorBase<Scalar>> 
+Teuchos::RCP<const Thyra::VectorBase<Scalar>>
 Piro::TempusIntegrator<Scalar>::getXDotDot() const
 {
-  Teuchos::RCP<const Thyra::VectorBase<Scalar>> xdotdot; 
+  Teuchos::RCP<const Thyra::VectorBase<Scalar>> xdotdot;
   if (basicIntegrator_ != Teuchos::null) {
-    xdotdot = basicIntegrator_->getXDotDot(); 
+    xdotdot = basicIntegrator_->getXDotDot();
   }
   if (fwdSensIntegrator_ != Teuchos::null) {
-    xdotdot = fwdSensIntegrator_->getXDotDot(); 
+    xdotdot = fwdSensIntegrator_->getXDotDot();
   }
   if (adjSensIntegrator_ != Teuchos::null) {
-    xdotdot = adjSensIntegrator_->getXDotDot(); 
+    xdotdot = adjSensIntegrator_->getXDotDot();
   }
-  return xdotdot; 
+  return xdotdot;
 }
 
 template <typename Scalar>
-Teuchos::RCP<const Tempus::SolutionHistory<Scalar>> 
+Teuchos::RCP<const Tempus::SolutionHistory<Scalar>>
 Piro::TempusIntegrator<Scalar>::getSolutionHistory() const
 {
-  Teuchos::RCP<const Tempus::SolutionHistory<Scalar>> soln_history; 
+  Teuchos::RCP<const Tempus::SolutionHistory<Scalar>> soln_history;
   if (basicIntegrator_ != Teuchos::null) {
-    soln_history = basicIntegrator_->getSolutionHistory(); 
+    soln_history = basicIntegrator_->getSolutionHistory();
   }
   if (fwdSensIntegrator_ != Teuchos::null) {
-    soln_history = fwdSensIntegrator_->getSolutionHistory(); 
+    soln_history = fwdSensIntegrator_->getSolutionHistory();
   }
   if (adjSensIntegrator_ != Teuchos::null) {
-    soln_history = adjSensIntegrator_->getSolutionHistory(); 
+    soln_history = adjSensIntegrator_->getSolutionHistory();
   }
-  return soln_history; 
+  return soln_history;
 }
 
 template <typename Scalar>
-Teuchos::RCP<const Tempus::TimeStepControl<Scalar>> 
+Teuchos::RCP<const Tempus::TimeStepControl<Scalar>>
 Piro::TempusIntegrator<Scalar>::getTimeStepControl() const
 {
-  Teuchos::RCP<const Tempus::TimeStepControl<Scalar>> ts_control; 
+  Teuchos::RCP<const Tempus::TimeStepControl<Scalar>> ts_control;
   if (basicIntegrator_ != Teuchos::null) {
-    ts_control = basicIntegrator_->getTimeStepControl(); 
+    ts_control = basicIntegrator_->getTimeStepControl();
   }
   if (fwdSensIntegrator_ != Teuchos::null) {
-    ts_control = fwdSensIntegrator_->getTimeStepControl(); 
+    ts_control = fwdSensIntegrator_->getTimeStepControl();
   }
   if (adjSensIntegrator_ != Teuchos::null) {
-    ts_control = adjSensIntegrator_->getTimeStepControl(); 
+    ts_control = adjSensIntegrator_->getTimeStepControl();
   }
-  return ts_control; 
+  return ts_control;
 }
 
 template <typename Scalar>
-void 
+void
 Piro::TempusIntegrator<Scalar>::clearObservers()
 {
   if (basicIntegrator_ != Teuchos::null) {
-    basicIntegrator_->getObserver()->clearObservers();
+    basicIntegrator_->setObserver();

```

Comment:
Is this going to change the timing in Albany?

Reply:
Hi @sconde!  Not I sure I understand.  Just switching to IntegratorBasic which holds the base class IntegratorObserver instead of a composite.  You can still give it a composite.  The IntegratorBasic just does not know about composite functionality.

Reply:
I'm worried about this issue: https://github.com/trilinos/Trilinos/issues/2163

Reply:
Gotcha.  I do not think this will be a problem here.

In IntegratorBasicOld, integratorObserver_ is an IntegratorObserverComposite and the setObserver(obs) function would **_add_** observers, obs, to integratorObserver_.   This means Piro needed to clearObservers() to make sure duplicates were not added.

In IntegratorBasic, integratorObserver_ is an IntegratorObserver and the setObserver(obs) function just sets the observer, obs, to integratorObserver_, while setObserver() (w/o args) sets integratorObserver_ to IntegratorObserverBasic.  So Piro needs to make sure it clears its IntegratorObserverComposite (which it appears they do), and then just set it on the IntegratorBasic.